pkg/cache/nodes.go

```diff
@@ -259,3 +258,12 @@ func (nc *schedulerNodes) schedulerNodeEventHandler() func(obj interface{}) {
 		}
 	}
 }
+
+func hasReadyCondition(node *v1.Node) bool {
+	for _, condition := range node.Status.Conditions {
```

Comment:
Can `node` or `Status` be nil? Rather be careful than rely on the externally defined object always being complete.